include/battle/battle_mon.h

```diff
@@ -0,0 +1,132 @@
+#ifndef POKEPLATINUM_BATTLE_MON_H
+#define POKEPLATINUM_BATTLE_MON_H
+
+#include "constants/moves.h"
+#include "constants/pokemon.h"
+#include "constants/string.h"
+
+typedef struct MoveEffectsData {
+    u32 disabledTurns           :3,
+        encoredTurns            :3,
+        chargedTurns            :2,
+        tauntedTurns            :3,
+        protectSuccessTurns     :2,
+        perishSongTurns         :2,
+        rolloutCount            :3,
+        furyCutterCount         :3,
+        stockpileCount          :3,
+        stockpileDefBoosts      :3,
+        stockpileSpDefBoosts    :3,
+        truantFlag              :1,
+        flashFireFlag           :1;
+    
+    u32 battlerIdLockOn     :2,
+        mimickedMoveSlot    :4,
+        battlerIdBinding    :2,
+        battlerIdMeanLook   :2,
+        lastResortCount     :3,
+        magnetRiseTurns     :3,
+        healBlockTurns      :3,
+        embargoTurns        :3,
+        hasHeldItem         :1, // this is specifically for tracking Unburden; 0 if no item was ever held, or if item is suppresed with Knock Off
+        metronomeTurns      :4,
+        micleBerryActive    :1,
+        custapBerryActive   :1,
+        quickClawActive     :1,
+        meFirstActive       :1,
+        padding0004_1F      :1;
+
+    int rechargeTurnCount;
+    int fakeOutTurnCount;
+    int slowStartTurnCount;
+    int meFirstTurnCount;
```

Comment:
Why not Turns like above?

Reply:
All the instances of `Turns` refer to a number of relative turns that get modified accordingly. All of these values refer to a *specific* turn on the counter. For example, whenever a mon comes into play, its `fakeOutTurnCount` is set to the current turn + 1, and that value is checked against the current turn whenever it tries to use Fake Out.

Reply:
I see, that makes more sense! Maybe `fakeOutRemainingTurns` or `fakeOutTurnsRemain`?

Reply:
`fakeOutTurnNumber`; the comparisons used for this field (and others like it) are always against the total turn counter in the parent `BattleContext` struct.